### DISCUSSION TITLE
Efficiently cropping / clipping TTGlyphs

### DISCUSSION THREAD

**Discussion by n8willis** [fonttools/fonttools#3026] [Q&A]
For analytical stuff, I routinely find myself cropping TTGlyphs — most often from, say, the baseline to the x-height, but occasionally on some other bounds, then continuing to do other stuff on the results.

Typically I've just done that by going through skia, with a pathPen etc., which is more than fine when it's interactive. But is there a more efficient way to do that, since it's a common operation? Is there a pen that can just do clipping?

(If so, I didn't see it and/or stopped looking because it wasn't that onerous with Pathops, but I wonder....)

**Comment by behdad** [comment]
I don't think there's an easier way than going through pathops, no.
